release/src/release-notes.ts

```diff
@@ -118,97 +120,97 @@ enum ProductCategory {
   other = "Other",
 }
 
-const getIssueType = (issue: Issue): IssueType => {
-  if (isNonUserFacingIssue(issue)) return IssueType.underTheHoodIssues;
-  if (isAlreadyFixedIssue(issue)) return IssueType.alreadyFixedIssues;
-  if (isBugIssue(issue)) return IssueType.bugFixes;
-  return IssueType.enhancements;
+type CategoryIssueMap = Record<Partial<ProductCategory>, Issue[]>;
+
+type IssueMap = {
+  [IssueType.bugFixes]: CategoryIssueMap;
+  [IssueType.enhancements]: CategoryIssueMap;
+  [IssueType.alreadyFixedIssues]: CategoryIssueMap;
+  [IssueType.underTheHoodIssues]: CategoryIssueMap;
 };
 
-const addIssueToCategory = (
-  issueMap: Record<IssueType, Partial<Record<ProductCategory, Issue[]>>>,
-  issue: Issue,
-  issueType: IssueType,
-  productCategory: ProductCategory,
-) => {
-  if (!issueMap[issueType][productCategory]) {
-    issueMap[issueType][productCategory] = [];
-  }
-  issueMap[issueType][productCategory]!.push(issue);
+const getIssueType = (issue: Issue): IssueType => {
+  return match(issue)
+    .when(isNonUserFacingIssue, () => IssueType.underTheHoodIssues)
+    .when(isAlreadyFixedIssue, () => IssueType.alreadyFixedIssues)
+    .when(isBugIssue, () => IssueType.bugFixes)
+    .otherwise(() => IssueType.enhancements);
 };
 
 const getLabels = (issue: Issue): string[] => {
   if (typeof issue.labels === "string") {
-    return [issue.labels];
+    return issue.labels.split(",");
   }
   return issue.labels.map(label => label.name || "");
 };
 
-const hasCategory = (labels: string[], category: string): boolean => {
-  return labels.some(label => label.includes(category));
+const hasCategory = (issue: Issue, categoryName: ProductCategory): boolean => {
+  const labels = getLabels(issue);
+  return labels.some(label => label.includes(categoryName));
 };
 
 export const getProductCategory = (issue: Issue): ProductCategory => {
-  const labels = getLabels(issue);
+  const category = Object.values(ProductCategory).find(categoryName =>
+    hasCategory(issue, categoryName)
+  );
 
-  if (hasCategory(labels, "Administration"))
-    return ProductCategory.administration;
-  if (hasCategory(labels, "Database")) return ProductCategory.database;
-  if (hasCategory(labels, "Embedding")) return ProductCategory.embedding;
-  if (hasCategory(labels, "Operation")) return ProductCategory.operation;
-  if (hasCategory(labels, "Organization")) return ProductCategory.organization;
-  if (hasCategory(labels, "Querying")) return ProductCategory.querying;
-  if (hasCategory(labels, "Reporting")) return ProductCategory.reporting;
-  if (hasCategory(labels, "Visualization"))
-    return ProductCategory.visualization;
```

Comment:
A big series of ifs like this is generally a smell. By using `Object.values` here we can drive all of our categorization off of that nice enum you defined, so adding/removing/changing categories will be a breeze.